src/sbd-inquisitor.c

```diff
@@ -56,6 +60,22 @@ void recruit_servant(const char *devname, pid_t pid)
 	newbie->pid = pid;
 	newbie->first_start = 1;
 
+	/* some sanity-check on our newbie */
+	if (sbd_is_disk(newbie)) {
+	    cl_log(LOG_INFO, "Monitoring %s", devname);
+	    disk_count++;
+	} else if (sbd_is_pcmk(newbie)) {
+	    newbie->outdated = 1;
+	} else if (sbd_is_cluster(newbie)) {
+	    newbie->outdated = 1;
+	} else {
+	    /* toss our newbie */
+	    cl_log(LOG_ERR, "Refusing to recrute unrecognized servant %s", devname);
```

Comment:
recruit

Reply:
Klaus, you didn't get what the `assert` in #67 was meant to
guard against primarily, and you've run into the very same
problem here with your new code that #67 was also meant to
avoid.  Remember, `strdup` can fail yielding `NULL` in that case.

Reply:
Sorry, it uses `devname`, not `newbie->devname`.

Reply:
yep, that prevents the null-pointer access. but you are right that an extra check with appropriate log might lead to more comprehensive logging.